Lesson 24

AndreyPushai opened the pull request:
Added test that checks adding, editing and deleting car in garage page.

AndreyPushai committed:
Upd registration.spec.ts

- added test.steps to positive test case

AndreyPushai committed:
Upd RegistrationPopUp.ts

- replaced deprecated type method with fill

AndreyPushai committed:
Upd POMs

- Added AddCarPopUp.ts
- Added CarItem.ts
- Added EditCarPopUp.ts
- Added GaragePage.ts
- Added RemoveCarPopUp.ts
- upd index.ts imports

AndreyPushai committed:
Upd dotenv file

AndreyPushai committed:
Added garage.spec.ts

- Added test, that checks create, edit and delete car functionalities

apolonskiy commented on poms/AddCarPopUp.ts:
```diff
@@ -0,0 +1,38 @@
+import BasePopUp from "./BasePopUp";
+
+
+export type CarModels = {
```
зазвичай T приписку до типу робимо
TCarModels

AndreyPushai replied:
Дякую, виправив.

apolonskiy commented on poms/EditCarPopUp.ts:
```diff
@@ -0,0 +1,35 @@
+    public removeCarButton = this.popUp.locator("button", {hasText: "Remove car"});
+
+    async selectBrand(
+        brand: "Audi" | "BMW" | "Ford" | "Porsche" | "Fiat"
```
чому б тут кіоф машинок теж не використати?

AndreyPushai replied:
Чудова ідея. Не розібрався з цією конструкцією generics ще. Тепер зрозумів. Переробив.

AndreyPushai committed:
Upd AddCarPopUp.ts and EditCarPopUp.ts

- Fixed name of type to TCarModels
- upd selectBrand methods with keyof TCarModels
